src/cli/onefuzz/backend.py

```diff
@@ -377,6 +378,8 @@ def container_file_path(container_url: str, blob_name: str) -> str:
 def serialize(data: Any) -> Any:
     if data is None:
         return data
+    if isinstance(data, SecretData) and not isinstance(data.secret, SecretAddress):
```

Comment:
Can we add some unit tests, to automate your example in the PR message?

Reply:
> This issue suggests `SecretData` should be more misuse-resistant. Is the deeper fix to also distinguish (at the type level) between saved and unsaved `SecretData`? It seems risky that the difference between serialization as plaintext and a gated reference is just whether or not a model has been saved.

I agree with the concern that `SecretData` needs work.  However, I am concern that will be a significantly larger issue.

Note, this is serialization used in the CLI backend, not SecretData in general.

I added a unittest that validates the functionality.